src/pages/DashboardAdmin/components/ManageHospital/components/HospitalDetailByAdmin.tsx

```diff
@@ -0,0 +1,263 @@
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { BackIcon } from "../../../styles/AddPersonnel.style";
+import ArrowLeft from "../../../../../resources/ArrowLeft.svg";
+import { useHistory, useParams } from "react-router";
+import { IHospital } from "../../../../../models/IHospital";
+import { ConfirmationModal } from "../../../../../components/ConfirmationModal";
+import { getHospital, updateHospital } from "../../../../../api/utilities";
+import Province from "../../../../RegisterHospitalAndAdmin/RegisterHospitalPage/province.json";
+
+type hospitalParams = {
+  id: string;
+};
+
+export const HospitalDetailByAdmin = () => {
+  const { id } = useParams<hospitalParams>();
+  let history = useHistory();
+  const [hospitalData, setHospitalData] = useState<IHospital>();
+  const [readOnly, setReadOnly] = useState<boolean>(true);
+  const [isShowModal, setIsShowModal] = useState(false);
+
+  const getParticularHospital = async () => {
+    const hospital = await getHospital(id);
+    setHospitalData(hospital.data);
+  };
+
+  const handleEditClick = () => {
+    setReadOnly(false);
+  };
+
+  const handleCancelClick = () => {
+    setReadOnly(true);
+    reset({
+      name: hospitalData?.name,
+      schedule: hospitalData?.schedule,
+      location: {
+        address: hospitalData?.location.address,
+        province: hospitalData?.location.province,
+      },
+      description: hospitalData?.description,
+    });
+  };
+
+  const goToPreviousPageClick = () => {
+    history.goBack();
+  };
+
+  const showModal = () => {
+    setIsShowModal(true);
+  };
+
+  const updateModal = (state: boolean) => {
+    setIsShowModal(state);
+  };
+
+  const { register, handleSubmit, reset } = useForm({
+    defaultValues: {
+      name: hospitalData?.name,
+      schedule: hospitalData?.schedule,
+      location: {
+        address: hospitalData?.location.address,
+        province: hospitalData?.location.province,
+      },
+      description: hospitalData?.description,
+    },
+  });
+
+  const onSubmit = async (data: any) => {
+    try {
+      const updateHospitalData = {
+        name: data.name,
+        schedule: data.schedule,
+        location: {
+          address: data.location.address,
+          province: data.location.province,
+        },
+        description: data.description,
+      };
+
+      if (updateHospitalData) {
+        await updateHospital(
+          hospitalData?.id.toString() as string,
+          updateHospitalData
+        ).then((res) => {
+          console.log(res);
+        });
+      }
+      updateModal(false);
+      window.location.reload();
```

Comment:
[Not a blocker] pero podemos evitar esto agregandole una dependencia al useEffect para que cargue de nuevo el hospital